src/Event/EventManager.php

```diff
@@ -371,11 +371,16 @@ public function dispatch($event)
         }
 
         $listeners = $this->listeners($event->name());
-        if (empty($listeners)) {
-            if ($this->_trackEvents) {
-                $this->addEventToList($event);
+
+        if ($this->_trackEvents) {
+            $this->addEventToList($event);
+
+            if (!$this->_isGlobal && static::instance()->isTrackingEvents()) {
```

Comment:
Could the global manager be tracking events while the local one is not? If so should this condition be outside the `if ($this->_trackEvents)` block?